fetch_worker.go

```diff
@@ -266,3 +284,136 @@ func (w *fetchWorker) buildFetchData(
 	key := snapshotKey{w.config.sortColName, keyVal}
 	return fetchData{key, w.config.table, payload, position}, nil
 }
+
+// fetchWorkerByLimit will perform a snapshot using the LIMIT + OFFSET clauses to fetch
+// rows in chunks.
+type fetchWorkerByLimit struct {
+	config fetchWorkerConfig
+	db     *sqlx.DB
+	data   chan fetchData
+	end    uint64
+}
+
+func newFetchWorkerByLimit(
+	db *sqlx.DB, data chan fetchData, config fetchWorkerConfig,
+) fetchWorker {
+	return &fetchWorkerByLimit{
+		db:     db,
+		data:   data,
+		config: config,
+	}
+}
+
+func (w *fetchWorkerByLimit) table() string {
+	return w.config.table
+}
+
+func (w *fetchWorkerByLimit) countTotal(ctx context.Context) (uint64, error) {
+	var total struct {
+		Total uint64 `db:"total"`
+	}
+
+	query, args, err := squirrel.Select("COUNT(*) as total").From(w.config.table).ToSql()
+	if err != nil {
+		return 0, fmt.Errorf("failed to build query: %w", err)
+	}
+
+	row := w.db.QueryRowxContext(ctx, query, args...)
+	if err := row.StructScan(&total); err != nil {
+		return 0, fmt.Errorf("failed to fetch total from %s: %w", w.config.table, err)
+	} else if err := row.Err(); err != nil {
+		return 0, fmt.Errorf("failed to fetch total from %s: %w", w.config.table, err)
+	}
+
+	sdk.Logger(ctx).Debug().Str("query", query).Any("args", args).Msg("count query")
```

Comment:
Really good that we have these logs :+1: If you don't mind, let's also add the number of rows found.

Reply:
Added the count result!